flagutil/flagutil.go

```diff
@@ -0,0 +1,90 @@
+// Package containing flag utilities. This includes implementations of flag.Value and flag.Getter interfaces. Also
+// contains methods like ValidateFlags.
+package flagutil
+
+import (
+	"flag"
+	"fmt"
+	"strings"
+)
+
+type FlagError struct {
+	Flag *flag.Flag
+	Err  error
+}
+
+func (f *FlagError) Error() string {
+	return fmt.Sprintf("Flag: %s, Error: %s, Usage: %s", f.Flag.Name, f.Err.Error(), f.Flag.Usage)
+}
+
+type MultiFlagError []*FlagError
+
+func (m MultiFlagError) Error() string {
+	errorstr := []string{"Errors found while validating flags"}
+	for _, err := range m {
+		errorstr = append(errorstr, err.Error())
+	}
+	return strings.Join(errorstr, "\n")
+}
+
+// Validates flags and logs a fatal message with all errors if found. Must be called after flag.Parse().
+// This is done by calling Get() on all flag values which implement the flag.Getter interface and collecting errors if returned.
+func ValidateFlags() MultiFlagError {
+	var flagErrors []*FlagError
+	checkGetError := func(f *flag.Flag) {
+		if fv, ok := f.Value.(flag.Getter); ok {
+			if err, ok := fv.Get().(error); ok && err != nil {
+				flagErrors = append(flagErrors, &FlagError{f, err})
+			}
+		}
+	}
+	flag.VisitAll(checkGetError)
+	if len(flagErrors) > 0 {
+		return MultiFlagError(flagErrors)
+	}
+	return nil
+}
+
+type flagValueError string
+
+func (err flagValueError) Error() string {
+	return string(err)
+}
+
+type requiredString struct {
+	str *string
+	set bool
+}
+
+// Adds a required string flag with the specified name. Note that ValidateFlags must be called to ensure that it is set.
+// Returns a pointer to the string value set.
+// Example usage:
+// var reqstr = flagutil.RequiredStringFlag(name, usage)
+// func main() {
+// 	flag.Parse()
+// 	if err := flagutil.ValidateFlags(); err != nil {
+// 		log.Fatal(err)
+// 	}
+// 	log.Println("reqstr")
+// }
```

Comment:
These comments are all really good. Only thing: can you keep the line lengths under 100 characters even in the comments?